### PR TITLE
implementation 2crud in front

### PR DESCRIPTION
**Pull request by Waleed2207**
implementation 2_full_cruds  for boards & tasks and export to csv in front.

### CODE REVIEW

File: controllers/boardsController.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    getBoard(req, res) {     
+        const result = boardsRepository.searchBoard(req.params.id);
+        if (result == null) {
+            const msg = 'Error! boarder no ' + req.params.id + ' was not found!';
+            Error(msg);
+            res.status(400).json(msg);
+        }
+        else {
+            logger.log("get board");
```
Comment (DaveAvigad) [inline controllers/boardsController.js:27]: redundant log

File: controllers/boardsController.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            res.status(200).json(result);
+        }  
+    },
+
+    updateBord(req, res) {
+        const ifExits = boardsRepository.searchBoard(req.body.id);
+        if(ifExits == null) {
```
Comment (DaveAvigad) [inline controllers/boardsController.js:34]: this if statement doesn't make sense. 
also, use ES6 compare type check 
improve readability.

File: controllers/boardsController.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        else {
+            boardsRepository.updateBoard(req.body);
+            logger.log("update a board");
+            res.status(200).json('Board no ' + req.body.id + ' was updated');
+        }
+    },
+
+    deleteBoard(req, res) {
+        console.log(req.params.id);
```
Comment (DaveAvigad) [inline controllers/boardsController.js:47]: too much redundant console.log

File: controllers/boardsController.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    createNewBoard(req, res) {
+        boardsRepository.createNewBoard(req.body);
+        logger.log("add a new board");
+        res.status(200).json('The board is created');
+    },
+    createNewTask(req,res) {
+        console.log(req.body)
+        boardsRepository.createNewTask(req.body);
+        logger.log("add a new tasks");
```
Comment (DaveAvigad) [inline controllers/boardsController.js:71]: why logger and console.log?

File: data/boards.json [hunk 5]
```diff
@@ -0,0 +1 @@
...lder than 30","Status":"To-Do","Priority":"Medium","Type":"Feature","Assignee":"roy_123@jmail.com","Creator":"guy_123@jmail.com"},{"TaskId":2,"TaskName":"Design","TaskDetails":"Prepare a table using React","Status":"In Progress","Priority":"Medium","Type":"Feature","Assignee":"amit_123@jmail.com","Creator":"guy_123@jmail.com"},{"TaskId":3,"TaskName":"waleed","TaskDetails":"software","Status":"To-do","Priority":"Low","Type":"Bug","Assignee":"waleed@gmail.com","Creator":"waled@gmail.com"}],"Employees":[{"EmployeeId":1},{"EmployeeId":2},{"EmployeeId":3},{"EmployeeId":4}]},{"BoardId":3,"BoardName":"java","Tasks":[{"TaskId":1,"TaskName":"t&y","TaskDetails":"fdg","Status":"low","Priority":"hight","Type":"feature","Assignee":"waleed@gmail.com","Creator":"ahmed@gmail.com"},{"TaskId":3,"TaskName":"q&u","TaskDetails":"software","Status":"In-progress","Priority":"Low","Type":"Bug","Assignee":"waleed@gmail.com","Creator":"waled@gmail.com"}],"Employees":[]},{"BoardId":4,"BoardName":"python","Tasks":[],"Employees":[]},{"BoardId":5,"BoardName":"crypto","Tasks":[],"Employees":[]},{"BoardId":6,"BoardName":"Linux","Tasks":[],"Employees":[]},{"BoardId":7,"BoardName":"wa#e","Tasks":[],"Employees":[]}]
```
Comment (DaveAvigad) [inline data/boards.json:1]: this data should be on the database

File: index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const cors = require('cors');
+const app = express();
+app.use(cors());
+const port = process.env.PORT || 3000;
+
+app.use(express.json());    
+app.use(express.urlencoded({extended: true}));  // מטפל בבקשות post עם body
```
Comment (DaveAvigad) [inline index.js:8]: remove comments (and never use Hebrew on the codebase)

File: index.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+
+const { boardsRouter } = require("./routers/boardsRouter");
+app.use('/api/boards', boardsRouter);
+
+app.use((req, res) => {
+    res.status(400).send('Something is broken!');
```
Comment (DaveAvigad) [inline index.js:15]: if something is broken it mosy likely needs to be status code 404?

File: logger/Logger.js [hunk 8]
```diff
@@ -6,6 +6,11 @@ const path = require('path');
 module.exports = class Logger extends EventEmitter{
     constructor() {
         super();
+        if (!Logger._instance) {
+            Logger._instance = this;
+        } else {
+            return Logger._instance;
```
Comment (DaveAvigad) [inline logger/Logger.js:12]: why the underscore?

File: node_modules/.bin/bunyan [hunk 9]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (DaveAvigad) [inline node_modules/.bin/bunyan:1]: do not commit `.node_modules`

File: repositories/boardsrepository.js [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    connectBoardJSON() {
+        const data = require('../data/boards.json');
+        this.setBoards(data);
+        this.on('updateBoards', () => {
+            fs.writeFile(Path.join(__dirname, '../data/boards.json'), JSON.stringify(this.data), 'utf8', err => {
+                if (err) 
+                    throw err;
+                console.log('File has been saved!');
```
Comment (DaveAvigad) [inline repositories/boardsrepository.js:20]: using logger or console.log? please do not mix

File: repositories/boardsrepository.js [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+            if (this.data[key].BoardId == id) {
+                return this.data[key];
+            }       
+        }
+        return null;
+    }
+    
+
+    updateBoard(payload) {
+        this.data.find(board => board.BoardId == payload.id).BoardName = payload.name;
```
Comment (DaveAvigad) [inline repositories/boardsrepository.js:49]: why `BoardName` is `uppecase`d?

File: repositories/boardsrepository.js [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+
+    createNewBoard(payload) {
+        let newID = 1;
+        if (this.data.length > 0)
+            newID = this.data[this.data.length - 1].BoardId + 1;
+
+        const newBoard = {
+            BoardId: newID,
```
Comment (DaveAvigad) [inline repositories/boardsrepository.js:60]: names shuold be lowercased

File: repositories/boardsrepository.js [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+    updateTaskByBoard(payload, boardId){
+        this.setTaskByBoard(boardId, [...this.data.find(board => board.BoardId == boardId).Tasks, payload]);
+        this.emit('updateBoards');
+    }
+    createNewTask(payload) {
+        let newID = 1;
+        if (this.data.find(board => board.BoardId == payload.BoardId).Tasks.length > 0)
+            newID = this.data.find(board => board.BoardId == payload.BoardId).Tasks[this.data.find(board => board.BoardId == payload.BoardId).Tasks.length - 1].TaskId + 1;
+        console.log(newID);
+        const newTask = {
```
Comment (DaveAvigad) [inline repositories/boardsrepository.js:87]: long functions,
too much code on one function 
consider decoupling it

File: repositories/boardsrepository.js [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+            TaskId: newID,
+            TaskName: payload.TaskName,
+            TaskDetails: payload.TaskDetails,
+            Status: payload.Status,
+            Priority: payload.Priority,
+            Type: payload.Type,
+            Assignee: payload.Assignee,
```
Comment (DaveAvigad) [inline repositories/boardsrepository.js:94]: why all the keys with the first uppercased letter?

File: repositories/boardsrepository.js [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+            this.data.find(board => board.BoardId == payload.BoardId).Tasks = this.data.find(board => board.BoardId == payload.BoardId).Tasks.filter(task => task.TaskId != payload.TaskId);
+            this.emit('updateBoards');
+        }
+        else {
+            this.emit('error');
+            return "error";
+        }
+    }
+    exportToCsv(boardId){
+        let ListToDo=this.data.find(board => board.BoardId == boardId).Tasks;
```
Comment (DaveAvigad) [inline repositories/boardsrepository.js:129]: this file doesn't use ESlint

File: routers/boardsRouter.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
+const { Router } = require('express');
+const { boardsController } = require('../controllers/boardsController');
+
+const boardsRouter = new Router();
```
Comment (DaveAvigad) [inline routers/boardsRouter.js:4]: great routes definition! 
according to RESTful api conventions!

File: src/js/boards.js [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+    });
+
+
+    for (let i = 0; i < insertemail.length; i++) {
+        insertemail[i].href += ("?email="+ userEmail);
+    }
+
+}
+
+const todayInput    = document.getElementById('dash-daterange');
```
Comment (DaveAvigad) [inline src/js/boards.js:59]: although the code is clean, it seems like too much repeated code

File: src/js/boards.js [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+        };
+    }
+}
+
+function editB() {
+    const editBoards    = document.getElementsByClassName('editboard');
+    for (let i = 0; i < editBoards.length; i++) {
+        editBoards[i].onclick = () => {
+            boardId = editBoards[i].childNodes[0].innerHTML;
+            editmsg.innerHTML = "Edit board number " + boardId + "."
```
Comment (DaveAvigad) [inline src/js/boards.js:96]: you are missing `;` here
consider using ESlint here

File: src/js/login.js [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+    const queryString = new URLSearchParams(paramString);
+    for (let pair of queryString.entries()) {
+      if(pair[0] == "boardId") {
+        boardId = pair[1];
+      }
+      if(pair[0] == "email") {
+        userEmail = pair[1];
+      }
+    }
+    // const user = getUser(userEmail);
```
Comment (DaveAvigad) [inline src/js/login.js:53]: remove comments

File: src/js/login.js [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+    });
+}
+
+const form      = document.getElementById('sign-in');
+const inputs    = document.getElementsByTagName('input');
+const error     = document.getElementById('error');
+
+function getUser(email) {
+    try {
+        const response =  fetch(`http://localhost:8081/api/login?email=${email}`);
```
Comment (DaveAvigad) [inline src/js/login.js:33]: move host and urls into constant file

File: src/js/tasks.js [hunk 21]
```diff
@@ -0,0 +1,272 @@
...
+const doughnut      = document.getElementById('doughnut');
+const csvBtn        = document.getElementById("csv");
+
+/*********************CreateTask**************************/
+
+const addmodeltask = document.getElementById('addtaskmodel');
+const addTasks     =document.getElementById('addtask');
+const newTaskName  = document.getElementById('taskname_new');
+const Taskdetails_new  = document.getElementById('taskdetails_new');
+const Status_new  = document.getElementById('status_new');
```
Comment (DaveAvigad) [inline src/js/tasks.js:140]: you are mixing between camelCase and snake_case variable names

**Comment by GavrieliYovel** [comment]
already implemented front from different  branch.

**Pull request closed without merging**